### PR TITLE
unskip hot plug block device and remove losetup EBUSY errors 

### PR DESCRIPTION
**Pull request by chavafg**

### CODE REVIEW

File: integration/docker/run_test.go [hunk 1]
```diff
@@ -112,13 +110,14 @@ func createLoopDevice() (string, string, error) {
...
-		losetupCmd := NewCommand("losetup", "-P", loopPath, f.Name())
-		_, _, exitCode := losetupCmd.Run()
-		if exitCode == 0 {
-			return f.Name(), loopPath, nil
-		}
+	losetupCmd := NewCommand("losetup", "-fP", f.Name())
+	_, _, exitCode := losetupCmd.Run()
+	if exitCode == 0 {
+		getLoopPath := NewCommand("losetup", "-j", f.Name())
+		stdout, _, _ := getLoopPath.Run()
```
Comment (devimc) [inline integration/docker/run_test.go]: I wonder if you should check the exit code

Comment (chavafg) [reply integration/docker/run_test.go]: fixed

File: integration/docker/run_test.go [hunk 2]
```diff
@@ -112,16 +110,20 @@ func createLoopDevice() (string, string, error) {
...
-	return "", "", fmt.Errorf("unable to create loop device for disk %s", f.Name())
+	// get loop device path
+	getLoopPath := NewCommand("losetup", "-j", f.Name())
+	stdout, stderr, exitCode := getLoopPath.Run()
+	if exitCode != 0 {
+		return "", "", fmt.Errorf("%s", stderr)
+	}
+	re := regexp.MustCompile("(.*?)\\:")
+	loopPath := re.FindStringSubmatch(stdout)
+	return f.Name(), loopPath[1], nil
```
Comment (devimc) [inline integration/docker/run_test.go]: uhmm , this will crash if `len(loopPath) < 2`

**Comment by devimc** [comment]
lgtm

File: integration/docker/run_test.go [hunk 3]
```diff
@@ -112,16 +110,23 @@ func createLoopDevice() (string, string, error) {
...
+	if _, stderr, exitCode := losetupCmd.Run(); exitCode != 0 {
+		return "", "", fmt.Errorf("%s", stderr)
 	}
 
-	return "", "", fmt.Errorf("unable to create loop device for disk %s", f.Name())
+	// get loop device path
+	getLoopPath := NewCommand("losetup", "-j", f.Name())
+	stdout, stderr, exitCode := getLoopPath.Run()
+	if exitCode != 0 {
+		return "", "", fmt.Errorf("%s", stderr)
```
Comment (jodh-intel) [inline integration/docker/run_test.go]: Nit: I'd add `stdout`, `stderr` and `exitCode` to the error for maximum debugability.

File: integration/docker/run_test.go [hunk 4]
```diff
@@ -112,16 +110,23 @@ func createLoopDevice() (string, string, error) {
...
+	}
+	re := regexp.MustCompile("/dev/loop[0-9]+")
+	loopPath := re.FindStringSubmatch(stdout)
+	if len(loopPath) == 0 {
+		return "", "", fmt.Errorf("Unable to get loop device path")
```
Comment (jodh-intel) [inline integration/docker/run_test.go]: Nit: I'd consider adding `stdout` and `stderr` [*] to the returned error to make it easier to see what went wrong.

---

[*] - which we'd *hope* was empty if `exitCode == 0`, but who knows? :)

**Pull request merged**